code/parachain/frame/composable-traits/src/xcm/assets.rs

```diff
@@ -89,3 +89,38 @@ pub struct ForeignMetadata<AssetNativeLocation> {
 	pub decimals: Option<Exponent>,
 	pub location: AssetNativeLocation,
 }
+
+pub trait PermissionlessAsset {
+	type AccountId;
+	type AssetId;
+	type AssetNativeLocation;
+	type Balance;
+
+	fn register_asset(
+		account_id: Self::AccountId,
+		asset_id: Self::AssetId,
+		location: Self::AssetNativeLocation,
+		name: Option<BiBoundedAssetName>,
+		symbol: Option<BiBoundedAssetSymbol>,
+		decimals: u8,
+		is_cosmwasm: bool,
+	) -> DispatchResult;
+}
+
+pub trait CosmWasmIssuance {
```

Comment:
Is not possible to use Substrate traits? I used Amm trait and it worked kind of well.

Reply:
i am ok with keep_alive parameters, etc